### PR TITLE
code: split out the handling of exceptions to a helper

### PR DESCRIPTION
**Pull request by keszybz**
This is a bit of error-handling logic that shouldn't obscure the main logic of the program.

### CODE REVIEW

**Comment by behrmann** [comment]
LGTM, I think `isort` wants an empty line between comment an first import.

File: mkosi/__main__.py [hunk 1]
```diff
@@ -1,5 +1,6 @@
 # SPDX-License-Identifier: LGPL-2.1+
 # PYTHON_ARGCOMPLETE_OK
+import contextlib
 import os
 import sys
 from subprocess import CalledProcessError
```
Comment (behrmann) [inline mkosi/__main__.py:6]:
```suggestion
from subprocess import CalledProcessError
from typing import Iterator
```

**Review by behrmann — COMMENTED** [review]
That should work

File: mkosi/__main__.py [hunk 2]
```diff
@@ -8,29 +9,33 @@
...
-            else:
-                run_verb(a)
+        yield
     except MkosiException as e:
         cause = e.__cause__
         if cause and isinstance(cause, CalledProcessError):
             sys.exit(cause.returncode)
         sys.exit(1)
 
+@propagate_failed_return()
```
Comment (behrmann) [inline mkosi/__main__.py:22]: Also another empty line for PEP8's sake
```suggestion

@propagate_failed_return()
```

File: mkosi/__main__.py [hunk 3]
```diff
@@ -1,5 +1,6 @@
 # SPDX-License-Identifier: LGPL-2.1+
 # PYTHON_ARGCOMPLETE_OK
+import contextlib
```
[the remark this answers is not in the document]
Comment (keszybz) [reply mkosi/__main__.py:3]: Why do we do this exercise?

File: mkosi/__main__.py (hunk 2 above)
Comment (keszybz) [reply mkosi/__main__.py:25]: Actually no, PEP8 as far as I remember, specified a single empty line between functions. Two empty lines was between classes. Overall, I find this enforced formatting just a waste of time.

**Review by DaanDeMeyer — APPROVED** [review]

File: mkosi/__main__.py (hunk 2 above)
Comment (behrmann) [reply mkosi/__main__.py:25]: It's [classes and top-level function definitions](https://peps.python.org/pep-0008/#blank-lines) and a single blank line between methods.

**Pull request merged**